### PR TITLE
improved error handling for pthread_create().

### PR DESCRIPTION
**Pull request by cubicdaiya**
Added the check for the return-value of `pthread_create()`.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -606,12 +606,16 @@ static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_
...
+            case RUN_MODE_WORKER: {
+                int err;
                 ssl_config->ocsp_stapling.interval =
                     ocsp_update_interval; /* is also used as a flag for indicating if the updater thread was spawned */
                 ssl_config->ocsp_stapling.max_failures = ocsp_max_failures;
-                pthread_create(&ssl_config->ocsp_stapling.updater_tid, NULL, ocsp_updater_thread, ssl_config);
-                break;
+                err = pthread_create(&ssl_config->ocsp_stapling.updater_tid, NULL, ocsp_updater_thread, ssl_config);
+                if (err != 0) {
+                    fprintf(stderr, "[OCSP Stapling] pthread_create for ocsp_updater_thread failed. errno:%d\n", err);
```
Comment (cubicdaiya) [inline src/main.c]: Even if `pthread_create()` for `ocsp_updater_thread()` fails, `h2o` can continue to serve. But I'm a little worried whether  `h2o` should continue to serve in this case. How about you?

**Comment by kazuho** [comment]
How aboout creating a wrapper function of `pthread_create` that emits an error and calls `abort(3)` in case it fails?  The function should be placed in include/h2o/multithread.h and lib/common/multithread.c, and possibly be named as `h2o_multithread_create_thread`.

The reasons I think we should go that way is:
* there are other places that call the function
* H2O is designed to only spawn a predetermined number of threads, which should all be created successfully

**Comment by cubicdaiya** [comment]
Hi, sorry for delay. Changed based on your opinion.

**Comment by kazuho** [comment]
Thank you for the changes.  Can you call `perror(3)` instead of `fprintf` + `strerror` for reporting the error?  `strerror` is not MT-safe, and should not be used once the server starts spawning threads.

**Comment by cubicdaiya** [comment]
Thank you for the suggestion. Pushed changes.

**Comment by kazuho** [comment]
Thank you!  merged to master.

**Pull request merged**